platforms/android/build_sdk.py

```diff
@@ -268,9 +268,9 @@ def build_library(self, abi, do_install, no_media_ndk):
 
         if self.hwasan and "arm64" in abi.name:
             hwasan_flags = "-fno-omit-frame-pointer -fsanitize=hwaddress"
-            cmake_vars['CMAKE_CXX_FLAGS_DEBUG']    = hwasan_flags
-            cmake_vars['CMAKE_C_FLAGS_DEBUG']      = hwasan_flags
-            cmake_vars['CMAKE_LINKER_FLAGS_DEBUG'] = hwasan_flags
+            for s in ['CMAKE_CXX_FLAGS_DEBUG',   'CMAKE_C_FLAGS_DEBUG',   'CMAKE_LINKER_FLAGS_DEBUG',
+                      'CMAKE_CXX_FLAGS_RELEASE', 'CMAKE_C_FLAGS_RELEASE', 'CMAKE_LINKER_FLAGS_RELEASE']:
```

Comment:
We should not touch generic CMake flags.

Use `OPENCV_EXTRA_` flags instead: https://github.com/opencv/opencv/blob/4.10.0/cmake/OpenCVCompilerOptions.cmake#L402-L418

Also append them instead of override if they already exist (e.g. defined by config file).

Reply:
@opencv-alalek I tuned the PR. Added flags to OPENCV_EXTRA_ and also enabled ASAN related quirks to fix false alarms.

Reply:
> Also append them instead of override if they already exist (e.g. defined by config file).